smdebug/tensorflow/keras.py

```diff
@@ -479,14 +518,15 @@ def _on_any_batch_begin(self, batch, mode, logs=None):
             self._prepare_collections()
 
         if self._prepared_tensors[mode] is False:
-            if self._validate_exec_function(self._get_exec_function(mode)):
+            if version.parse(tf.__version__) >= version.parse(
+                "2.0.0"
+            ) or self._validate_exec_function(self._get_exec_function(mode)):
```

Comment:
or -> and?

Reply:
"or" 
validate exec function is valid only for 1.x versions. prepare_layers can be be called only if this validate call returns a non-None value.